Kconfig.nrf

```diff
@@ -8,6 +8,12 @@ rsource "subsys/net/openthread/Kconfig.defconfig"
 
 menu "Nordic nRF Connect"
 
+# Override configuration from zephyr which sets this to 0x200 if MCUboot is
+# enabled (CONFIG_BOOTLOADER_MCUBOOT), since NCS use partition_manager to
+# get this offset intsead.
+config ROM_START_OFFSET
+	default 0 if PARTITION_MANAGER_ENABLED
```

Comment:
looks good to me, but would like confirmation from @nordicjm that this doesn't impact current PM support work in sysbuild (or that at least that work can adjust the `if` part if needed.